client/normal_value.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package client
+
+import (
+	"time"
+
+	"github.com/sourcenetwork/immutable"
+)
+
+// NormalValue is the interface for the normal value types.
+// It is used to represent the normal (or standard) values across the system and to avoid
+// asserting all possible types like int, int32, int64, etc.
+//
+// All methods returning a specific type returns the value and the second boolean flag indicating
+// if the value is of the requested type.
+//
+// All nillable values are represented as immutable.Option[T].
+type NormalValue interface {
```

Comment:
suggestion: A `Kind() FieldKind` func might be handy to have on this.

Reply:
good suggestion but we don't have all `FieldKind` to cover all possible normal values. For example, we don't have `FieldKind_INT`.

Reply:
I think that can be sorted out when we add support for `FieldKind_INT`

Reply:
this is somewhat involved addition. I've created another task for it https://github.com/sourcenetwork/defradb/issues/2463